### PR TITLE
[bugfix][hail] Capture error rather than printing it

### LINKED ISSUE #8146: VEP output going to stderr
See discuss post: https://discuss.hail.is/t/redirect-or-find-vep-or-other-error-output-from-a-hail-pipeline/1308/9?u=danking

It looks like Hail isn’t capturing all the VEP output. Can someone look into this? Probably the way we’re executing external commands needs to also capture stderr and print it.

Assigning Tim for delegation.

### PR DESCRIPTION
**Pull request by chrisvittal**
This is untested, but probably can't make things worse.

fixes #8146

### CODE REVIEW

**Comment by johnc1231** [comment]
I think we need to at least manually test it.

File: hail/src/main/scala/is/hail/methods/VEP.scala [hunk 1]
```diff
@@ -67,14 +67,12 @@ object VEP {
...
-  def waitFor(proc: Process, cmd: Array[String]): Unit = {
+  def waitFor(proc: Process, err: StringBuilder, cmd: Array[String]): Unit = {
     val rc = proc.waitFor()
 
     if (rc != 0) {
-      val errorLines = Source.fromInputStream(new BufferedInputStream(proc.getErrorStream)).getLines().mkString("\n")
-
       fatal(s"VEP command '${ cmd.mkString(" ") }' failed with non-zero exit status $rc\n" +
-        "  VEP Error output:\n" + errorLines)
+        "  VEP Error output:\n" + err.toString)
```
Comment (danking) [inline hail/src/main/scala/is/hail/methods/VEP.scala]: Can you help me understand why this works when `proc.getErrorStream` doesn't? I expected we would have an err stream and an out stream and we would include both (separately) in the exception message.

Comment (chrisvittal) [reply hail/src/main/scala/is/hail/methods/VEP.scala]: The output stream is VEP's output, which we parse. The issue here is with the `pipe` method, changed later in this PR. We set up a thread to read from VEP's stderr pipe. We can only read from that stream once. Previously we were printing it to the JVM's `System.err` stream, now, we're capturing it in a `StringBuilder`.

**Comment by chrisvittal** [comment]
How do I test this? I don't know how to make vep fail.

**Comment by johnc1231** [comment]
I just meant we should test that VEP still works on this branch. Like, do we have automated tests for VEP?

**Comment by chrisvittal** [comment]
Oh, that's easy, `make test-dataproc`, doing that now.

**Comment by chrisvittal** [comment]
@johnc1231 `make test-dataproc` succeeded

**Pull request merged**